subsys/fs/fs.c

```diff
@@ -148,6 +148,10 @@ int fs_open(struct fs_file_t *zfp, const char *file_name, fs_mode_t flags)
 	}
 
 	zfp->mp = mp;
+	if (mp->flags & FS_MOUNT_FLAG_READ_ONLY &&
```

Comment:
A reason combining multiple bits into the named flag is bad: this test will pass if `FS_MOUNT_FLAG_NO_FORMAT` is the only mount flag, which would be wrong.

Reply:
It's also an argument for what MISRA would require:

    ((mp->flags & FS_MOUNT_FLAG_READ_ONLY) == FS_MOUNT_FLAG_READ_ONLY)

would work where

    ((mp->flags & FS_MOUNT_FLAG_READ_ONLY) != 0)

would not.

Reply:
Bummer. Will fix.